resources/sdk/purecloudjavascript/templates/ApiClient.mustache

```diff
@@ -417,6 +417,339 @@ class ApiClient {
+												redirect_uri: redirectUri })
+			});
+
+			request.proxy = this.proxy;
+			var bodyParam = {
+				grant_type: 'authorization_code',
+                code: authCode,
+                code_verifier: codeVerifier,
+                client_id: clientId,
+                redirect_uri: redirectUri,
+			};
+			// Handle response
+			request
+			.then((response) => {
+				// Logging
+				this.config.logger.log(
+					'trace',
+					response.status,
+					'POST',
+					`https://login.${this.config.environment}/oauth/token`,
+					request.headers,
+					response.headers,
+					bodyParam,
+					undefined
+				);
+				this.config.logger.log(
+					'debug',
+					response.status,
+					'POST',
+					`https://login.${this.config.environment}/oauth/token`,
+					request.headers,
+					undefined,
+					bodyParam,
+					undefined
+				);
+
+				// Get access token from response
+				var access_token = response.data.access_token;
+
+				this.setAccessToken(access_token);
+				this.authData.tokenExpiryTime = new Date().getTime() + response.data['expires_in'] * 1000;
+				this.authData.tokenExpiryTimeString = new Date(this.authData.tokenExpiryTime).toUTCString();
+
+				// Return auth data
+				resolve(this.authData);
+			})
+			.catch((error) => {
+				// Log error
+				if (error.response) {
+					this.config.logger.log(
+						'error',
+						error.response.status,
+						'POST',
+						`https://login.${this.config.environment}/oauth/token`,
+						request.headers,
+						error.response.headers,
+						bodyParam,
+						error.response.data
+					);
+				}
+
+				reject(error);
+			});
+
+		});
+	}
+
+	/**
+	 * @description Generate a random string used as PKCE Code Verifier - length = 43 to 128.
+	 * @param {number} nChar - code length
+	 */
+	generatePKCECodeVerifier(nChar) {
+		if (nChar < 43 || nChar > 128) {
+			throw new Error(`PKCE Code Verifier (length) must be between 43 and 128 characters`);
+		}
+		// Check for window
+		if (typeof window === 'undefined') {
+			try {
+				const getRandomValues = require('crypto').getRandomValues;
+				const unreservedCharacters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-._~";
+				let randomString = Array.from(getRandomValues(new Uint32Array(nChar)))
+					.map((x) => unreservedCharacters[x % unreservedCharacters.length])
+					.join('');
+				return randomString;
+			} catch (err) {
+				// do nothing
+			}
+		} else {
+			const unreservedCharacters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-._~";
+			let randomString = Array.from(crypto.getRandomValues(new Uint32Array(nChar)))
+				.map((x) => unreservedCharacters[x % unreservedCharacters.length])
+				.join('');
+			return randomString;
+		}
+	}
+
+	/**
+	 * @description Compute Base64Url PKCE Code Challenge from Code Verifier.
+	 * @param {string} code - code verifier used to generate the code challenge
+	 */
+	computePKCECodeChallenge(code) {
+		if (code.length < 43 || code.length > 128) {
```

Comment:
You might need to check that code is defined, I was able to get this:
```
if (code.length < 43 || code.length > 128) {
                         ^
TypeError: Cannot read properties of undefined (reading 'length')
```
A defined error message would make it easier to know what is happening